### PR TITLE
[FIX] Correspondence Analysis: do not crash when no categorical

### PR DESCRIPTION
**Pull request by jerneju**
##### Issue
Widget crashes when there are no categorical variables in a data.

##### Description of changes
Check and show error.

##### Includes
- [X] Code changes
- [x] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by jerneju** [comment]
https://sentry.io/biolab/orange3/issues/319091508/

File: Orange/widgets/unsupervised/owcorrespondence.py [hunk 1]
```diff
@@ -61,6 +61,7 @@ class Inputs:
 
     class Error(widget.OWWidget.Error):
         empty_data = widget.Msg("Empty data set")
+        no_disc_vars = widget.Msg("Correspondence Analysis needs categorical variables")
```
Comment (janezd) [inline Orange/widgets/unsupervised/owcorrespondence.py]: Add a period at the end (we follow the rule that if the message is a complete sentence, we add a period).

Alternatively, shorten this to `"No categorical variables"` (without a period).

File: Orange/widgets/unsupervised/owcorrespondence.py [hunk 2]
```diff
@@ -113,12 +115,14 @@ def set_data(self, data):
...
             self.varlist[:] = [var for var in data.domain.variables
                                if var.is_discrete]
-            self.selected_var_indices = [0, 1][:len(self.varlist)]
-            self.component_x = 0
-            self.component_y = int(len(self.varlist[self.selected_var_indices[-1]].values) > 1)
-            self.openContext(data)
-            self._restore_selection()
-#             self._invalidate()
+            if not len(self.varlist[:]):
+                self.Error.no_disc_vars()
```
Comment (janezd) [inline Orange/widgets/unsupervised/owcorrespondence.py]: Add `self.data = None` here. Not needed right now, but the above code sets it to `None` as the indicator that no data is shown.

File: Orange/widgets/unsupervised/tests/test_owcorrespondence.py [hunk 3]
```diff
@@ -51,3 +51,20 @@ def test_data_one_value_zero(self):
...
+            Domain(
+                [ContinuousVariable("a")]
+            ),
+            [(1,), (2,), (3,)]
+        )
+        self.assertFalse(self.widget.Error.no_disc_vars.is_shown())
+        self.send_signal(self.widget.Inputs.data, table)
+        self.assertTrue(self.widget.Error.no_disc_vars.is_shown())
+        self.send_signal(self.widget.Inputs.data, None)
+        self.assertFalse(self.widget.Error.no_disc_vars.is_shown())
```
Comment (janezd) [inline Orange/widgets/unsupervised/tests/test_owcorrespondence.py]: Add

```
        self.send_signal(self.widget.Inputs.data, table)
        self.assertTrue(self.widget.Error.no_disc_vars.is_shown())
        self.send_signal(self.widget.Inputs.data, Table("iris"))
        self.assertFalse(self.widget.Error.no_disc_vars.is_shown())
```

We have had bugs in the past where the error message was sometimes not cleared.

File: Orange/widgets/unsupervised/owcorrespondence.py [hunk 4]
```diff
@@ -102,6 +103,7 @@ def __init__(self):
     def set_data(self, data):
         self.closeContext()
         self.clear()
+        self.Error.no_disc_vars.clear()
```
Comment (janezd) [inline Orange/widgets/unsupervised/owcorrespondence.py]: Since the widget is reset here, I suppose that no error should remain visible from the previous data. So you can safely call `self.Error.clear()` instead of `self.Error.no_disc_vars.clear()`, and then you can also remove `else: self.Error.empty_data.clear()` below.

**Comment by janezd** [comment]
Actually, I can take care of my nitpicking myself. :)
